src/board.rs

```diff
@@ -228,6 +299,115 @@ impl Chessboard {
         // Return the FEN string
         string_array.join(" ")
     }
+
+    /* *************** */
+    /* PRIVATE FUNCTIONS */
+
+    /// Maps the pieces on the chessboard to their character representations in the console.
+    ///
+    /// # Returns
+    ///
+    /// A vector of tuples, where each tuple consists of a chess piece character and its
+    /// corresponding bitboard positions. The characters represent different chess pieces,
+    /// and the bitboard positions indicate the squares occupied by those pieces on the board.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use kn_o3::Chessboard;
+    ///
+    /// let initial_position = Chessboard::new();
+    /// let pieces_mapping = initial_position.get_pieces();
+    /// for (piece_char, positions) in pieces_mapping {
+    ///     println!("Piece: {} | Bitboard Positions: {}", piece_char, positions);
+    /// }
+    /// ```
+    ///
+    /// The function returns a vector containing tuples, each associating a chess piece
+    /// character ('P', 'N', 'B', 'K', 'Q', 'R', 'p', 'n', 'b', 'k', 'q', 'r') with its
+    /// corresponding bitboard positions on the chessboard.
+    fn get_pieces(&self) -> Vec<(char, u64)> {
+        vec![
+            ('P', self.white_pawns),
+            ('N', self.white_knights),
+            ('B', self.white_bishops),
+            ('K', self.white_king),
+            ('Q', self.white_queen),
+            ('R', self.white_rooks),
+            ('p', self.black_pawns),
+            ('n', self.black_knights),
+            ('b', self.black_bishops),
+            ('k', self.black_king),
+            ('q', self.black_queen),
+            ('r', self.black_rooks),
+        ]
+    }
```

Comment:
I know I wrote this function, but we should change this to an array since we know it's always going to be 12 long.
(Can you store tuples inside arrays?!?!)
Prefer stack over heap.

Reply:
Let's make an issue for this